### PR TITLE
Fixes to the Solo for Demo

### LINKED ISSUE #3829: ag-solo from `agoricstage-15` does not wake up after provisioning
## Describe the bug

After doing a `docker compose up` I get something like this:

```
Attaching to ag-solo_1
ag-solo_1  | 2021-09-15T05:18:13.499Z chain-cosmos-sdk: 
ag-solo_1  | =============
ag-solo_1  | agoricstage-15 chain does not yet know of address agoric12ugfd4dgj5j6qv9nudys6r4ykujcjzvpzjhlka
ag-solo_1  | 
ag-solo_1  | Send:
ag-solo_1  | 
ag-solo_1  |   !faucet client agoric12ugfd4dgj5j6qv9nudys6r4ykujcjzvpzjhlka
ag-solo_1  | 
ag-solo_1  | to the appropriate faucet channel on Discord (https://agoric.com/discord)
ag-solo_1  | =============
ag-solo_1  | 
```

Then, even though I provision the client, it doesn't wake up.  Apparently the wallet is attempting to deploy in the meantime, but it just prints a bunch of `ooooo` and never continues.

## Expected behavior

Client should wake up right away after provisioning.

It does properly wake up if the ag-solo is restarted.

## Platform Environment
Docker compose as per instructions.

## Additional context

The fact that the wallet is trying to deploy even before we're provisioned is suspicious.  It may be necessary to have the `chain-cosmos-sdk.js` await the first message from the remote before returning control to the `solo`.

### PR DESCRIPTION
**Pull request by michaelfig**
Fixes #3829

@dckc  The `fix(solo)` is the important one (where storage events on transactions weren't being detected).  The rest is minor tidying of `agoric-cli` and `deployment`.

@samsiegart please look at the `dapp-svelte-wallet/ui/src/main.js` change to default to the V2 UI, now that you've got it feature-complete.

### CODE REVIEW

File: packages/dapp-svelte-wallet/ui/src/main.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ import App from './App.svelte';
 import AppV2 from './AppV2.svelte';
 
 const urlParams = new URLSearchParams(window.location.search);
-const isV2 = urlParams.get('ui') === 'v2';
+const isV2 = urlParams.get('ui') !== 'v1';
```
Comment (samsiegart) [inline packages/dapp-svelte-wallet/ui/src/main.js]: I'm okay with making it the default as long as you agree it's in a ready enough state for all users. We can always manually add v2 for the demo, I was just hesitant about enabling it for all users in an unfinished state. Not that it's not already an improvement over v1, but I didn't want people to look at it expecting it to be fully done and wonder why there's still a lot of rough edges.

**Comment by samsiegart** [comment]
LGTM but I'll defer to Dan for the rubber stamp

**Comment by dckc** [comment]
Would you please sketch the steps to reproduce the `NewBlockHeader` problem?

**Review by dckc — APPROVED** [review]

**Comment by michaelfig** [comment]
Testing the ag-solo fix:

```sh
cd packages/cosmic-swingset
make scenario2-setup scenario2-run-chain
# Another window:
make wait-for-cosmos t1-start-ag-solo
# Another window after the solo is waiting for provisioning:
make t1-provision-with-powers
```

Then the solo should detect the provisioning and start sending messages.  Before the fix was landed, the solo just waits printing 'o's.

File: packages/dapp-svelte-wallet/ui/src/main.js (hunk 1 above)
Comment (michaelfig) [reply packages/dapp-svelte-wallet/ui/src/main.js]: > Not that it's not already an improvement over v1, but I didn't want people to look at it expecting it to be fully done and wonder why there's still a lot of rough edges.

Fair enough.  It's reasonable to add the `?ui=v2`, so I put that in the demo script doc.

**Pull request merged**